src/java.base/share/classes/jdk/internal/classfile/impl/verifier/ParserVerifier.java

```diff
@@ -201,17 +201,19 @@ private void verifyAttributes(ClassFileElement cfe, List<VerifyError> errors) {
     }
 
     private void verifyAttribute(AttributedElement ae, Attribute<?> a, List<VerifyError> errors) {
-        int payLoad = ((BoundAttribute)a).payloadLen();
-        if (payLoad != switch (a) {
-            case AnnotationDefaultAttribute aa -> valueSize(aa.defaultValue());
+        int size = -1;
+        switch (a) {
```

Comment:
Maybe use a switch expression to set `size` where `default` yields -1?

Reply:
For better readability I've recently turned it from switch expression (yielding values pre-calculated in temporary locals) into explicit assignments to the `size` variable.

Reply:
I've turned it back to switch expression and compacted it a bit more.